django/forms/models.py

```diff
@@ -203,6 +204,10 @@ def fields_for_model(
                 fields is not None
                 and f.name in fields
                 and (exclude is None or f.name not in exclude)
+                # Declaring a model form field for a GFK worked until GFK
+                # inherited from Field in commit 6002df0, causing a regression
+                # with FieldError. This line allows it again:
+                and not isinstance(f, GenericForeignKey)
```

Comment:
As for me, this patch is not acceptable, `forms` should not be aware of `contrib` apps. Any fix (if needed) should be in `django.contrib.contenttypes.fields`, not here.

Reply:
Turns out we have almost identical behavior being tested in `generic_relations_regress.tests.GenericRelationTests.test_editable_generic_rel` for `GenericRelation`.
Here there is a `SpecialGenericRelation` defined which updates the `__init__` to have `self.editable = True`. 

As this is the same work around as what was suggested in the [Django forum discussion](https://forum.djangoproject.com/t/non-editable-error-for-a-genericforeignkey-in-an-modelform-when-updating-to-version-5-1/35033/4), this makes me think that we should `wontfix` the ticket and point to the forum discussion which "blesses" this approach.

Reply:
Also an option, probably too creative: if we set editable=True by default, we can still raise a NotImplementedError in the formfield() method so that the field can't by used in a form without a form field declaration, we could have `__all__` skip fields when formfield raises NotImplementedError?

If we can't find a way to restore the behavior,  shouldn't we then amend the release notes to document the BC break?

Reply:
I pushed another change which might be acceptable 🤔

Reply:
I think it makes sense to land this for 5.1 and to create a new ticket which will update the API to support defining `GenericForeignKey(..., editable=True)` as a new feature and also deprecate this error bypassing